radicale_imap/__init__.py

```diff
@@ -54,17 +82,11 @@ def is_authenticated(self, user, password):
         if sys.version_info < (3, 4) and secure:
             raise RuntimeError("Secure IMAP is not availabe in Python < 3.4")
         try:
-            connection = imaplib.IMAP4(host=address, port=port)
-            try:
-                if sys.version_info < (3, 4):
-                    connection.starttls()
-                else:
-                    connection.starttls(ssl.create_default_context())
-            except (imaplib.IMAP4.error, ssl.CertificateError) as e:
-                if secure:
-                    raise
-                self.logger.debug("Failed to establish secure connection: %s",
-                                  e, exc_info=True)
+            if secure or imaps:
+                connection = self.get_secure_connection(address, port, imaps)
+            else:
+                connection = self.get_connection(address, port)
```

Comment:
If ``secure`` is false we can still try to use STARTTLS opportunistically, to protect against passive monitoring.

Reply:
I strongly disagree, this is a tool for people who know what they are doing. If I explicitly disable transport encryption I expect that it honors that, the current implementation is broken btw, it fails if STARTTLS is not available.
If at all opportunistic encryption should have it's own switch.